src/core/qgsdistancearea.cpp

```diff
@@ -384,6 +400,12 @@ QgsPointXY QgsDistanceArea::computeSpheroidProject(
   double a = mSemiMajor;
   double b = mSemiMinor;
   double f = 1 / mInvFlattening;
+  if ( ( ( a < 0 ) && ( b < 0 ) ) ||
+       ( ( p1.x() < -180.0 ) || ( p1.x() > 180.0 ) || ( p1.y() < -85.05115 ) || ( p1.y() > 85.05115 ) ) )
```

Comment:
I would say the given value should be retained
* since the nearer you get to a pole, the returned result are not correct

SELECT ST_Project(MakePoint(-180,-90,4326),1,1.570796327);
SRID=4326;POINT(-90.00000001062518 -89.99999104696597)

SELECT ST_Distance(MakePoint(-180,-85.05115,4326),ST_Project(MakePoint(-180,-85.05115,4326),1,1.570796327))
* 0.000104  in mapunits [**correct**]

SELECT ST_Distance(MakePoint(-180,-86.05115,4326),ST_Project(MakePoint(-180,-86.05115,4326),1,1.570796327))
* 0.000130  in mapunits [**correct**]

SELECT ST_Distance(MakePoint(-180,-86.05115,4326),ST_Project(MakePoint(-180,-87.05115,4326),1,1.570796327))
* 1.000000  in mapunits [**false**]

SELECT ST_Distance(MakePoint(-180,-90,4326),ST_Project(MakePoint(-180,-90,4326),1,1.570796327))
* 90 in mapunits [**false**]